doc/rtd/reference/datasources/wsl.rst

```diff
@@ -0,0 +1,173 @@
+   the Windows host filesystem.
+
+3. **The init system must be aware of cloud-init**. WSL has opt-in support for
+   systemd, thus for distros that rely on it, such as Ubuntu, cloud-init will
+   run automatically if systemd is enabled via the ``/etc/wsl.conf``. The
+   Ubuntu applications distributed via Microsoft Store enable systemd in the
+   first boot, so no action is required if the user sets up a new instance by
+   using them. Users of other distros may find surprising that cloud-init
+   doesn't run automatically by default. At the time of this writing only
+   systemd distros are supported by the WSL datasource, although there is
+   nothing hard coded in the implementation code that requires it, so it's
+   possible that non-systemd distros find ways to run cloud-init and make it
+   just work.
+
+Notice that requirements 1 and 2 are met by default, i.e. WSL grants those
+features enabled, being the user allowed to opt out, if so desired.
+For more information about how to configure WSL,
+`check the official documentation <https://learn.microsoft.com/windows/wsl/wsl-config#configuration-settings-for-wslconf>`_.
+
+User data Configuration
+========================
+
+The WSL datasource relies exclusively on the Windows filesystem as the provider
+for user data. Access to those files is provided by WSL itself, unless disabled
+by the user, thus the datasource doesn't require any special component running
+on the Windows host to provide such data.
+
+User data can be supplied in any
+:ref:`format supported by cloud-init<user_data_formats>`, such as YAML
+cloud-config files or shell scripts. At runtime the WSL datasource looks for
+user data in the following locations inside the Windows host filesystem, in the
+order specified below:
+
+1. ``%USERPROFILE%\.cloud-init\<InstanceName>.user-data`` holds user data for a
+   specific instance configuration. The datasource resolves the name attributed
+   by WSL to the instance being initialized and looks for this file before any
+   of the subsequent alternatives. Example: ``sid-mlkit.user-data`` matches an
+   instance named ``Sid-MLKit``.
+
+2. ``%USERPROFILE%\.cloud-init\<ID>-<VERSION_CODENAME>.user-data`` for the
+   distro specific configuration, matched by the distro ID and VERSION_CODENAME
+   entries as specified in ``/etc/os-release``. Example:
+   ``ubuntu-jammy.userdata`` will affect any instance created from an Ubuntu
+   22.04 Jammy Jellyfish image, if a more specific configuration file does not
+   match.
+
+3. ``%USERPROFILE%\.cloud-init\<ID>-all.user-data`` for the distro specific
+   configuration, matched by the distro ID entry in ``/etc/os-release``,
+   regardless of the release version. Example: ``debian-all.user-data`` will
+   affect any instance created from any Debian GNU/Linux image, regardless of
+   which release, if a more specific configuration file does not match.
+
+4. ``%USERPROFILE%\.cloud-init\config.user-data`` for the configuration
```

Comment:
When WSL interoperability and automount is enabled, will the WSL agent provide this provide this empty clonfig.user-data file by default to all USERPROFILE? Or, does the user always need to set this default up so prior to using cloud-init?

Reply:
We may want to just rename this ``%USERPROFILE%\.cloud-init\all.user-data``  to also align with the ``<DISTRO_ID>-all.user-data`` encoded filename above or maybe ``%USERPROFILE%\.cloud-init\default.user-data``.  It seems strange to pivot from ``<DISTRO_ID>-all.user-data`` to ``config.user-data`` as the default.

Reply:
`%USERPROFILE%\.cloud-init\default.user-data` looks better. I'm going with that suggestion.